src/boundaries/stagger_and_step.jl

```diff
@@ -0,0 +1,250 @@
+      `x` such that `x_c = x + u*r`. `r = 10^-s` with `s` taken
+      from a uniform distribution in [-15, δ]. 
+
+    * `:unif`: The next candidate is `x_c = x + u*r` with `r` 
+      taken from a uniform distribution [0,δ] and `u` a random 
+      direction around `x`. 
+
+    * `:adaptive`: The next candidate is `x_c = x + u*r` with 
+      `r` drawn from a gaussian distribution with variance  δ 
+      and mean zero.
+      The variance is adapted according to a free parameter `γ`
+      such that: `δ = δ/γ` if no candidate is found and `δ = δ*γ`
+      when it succeeds. 
+
+* `γ = 1.1`: It is free parameter for the adaptive stagger method `:adaptive`. 
+
+* `δ₀ = 1.0`: This is the radius for the first stagger 
+  trajectory search. The algorithm looks for a point 
+  sufficiently close to the saddle before switching to the 
+  stagger-and-step routine. The search radius must be large 
+  enough to find a suitable initial candidate. 
+
+* `rng::AbstractRNG = MersenneTwister()`:  Random number generator. Use this for
+  reproducibility.
+
+## Description 
+
+The method relies on the stagger-and-step algorithm that 
+search initial conditions close to the saddle with escapes time 
+`T(x_n) > Tm`. The function `T` represents the iteration number 
+at which the trajectory with initial condition `x_n` steps out 
+from a region defined by the user (see the argument `isinside`).
+
+Given the dynamical mapping `F`, if the iteration `x_{n+1} = 
+F(x_n)` respects the condition `T(x_{n+1}) > Tm` we accept 
+this next point, this is the _step_ part of the method. If 
+not, the method search randomly the next point in a 
+neighborhood following a given probability distribution, this 
+is the _stagger_ part. This part sometimes fails to find a new 
+candidate and a new starting point of the trajectory is chosen 
+within the defined region. See the keyword argument 
+`stagger_mode` for the different available methods.   
+
+The method produces a pseudo-trajectory of `N` points δ-close
+to the stable manifold of the chaotic saddle. 
+
+"""
+function stagger_and_step!(ds::DynamicalSystem, x0, N::Int, isinside::Function; δ = 1e-10, Tm  = 30, 
+    γ = 1.1, max_steps = Int(1e5), max_escape_time = 10000, stagger_mode = :exp, δ₀ = 1., 
+    show_progress = true,  rng = MersenneTwister())
+
+    progress = ProgressMeter.Progress(
+        N; desc = "Saddle estimation: ", dt = 1.0
+    )
+    xi = stagger_trajectory!(ds, x0, Tm, isinside; δ₀, stagger_mode = :unif, 
+                            max_steps, γ, max_escape_time, rng) 
+    if isnothing(xi)
+        error("Cannot find a stagger trajectory. Choose a different starting point or 
+                search radius δ₀.")
+    end
+
+    v = Vector{typeof(current_state(ds))}(undef, N)
+    v[1] = xi
+    for n in 1:N
+        show_progress && ProgressMeter.update!(progress, n)
+        if escape_time!(ds, xi, isinside; max_escape_time) > Tm
+            reinit!(ds, xi; t0 = 0)
```

Comment:
Why is here `t0` imposed as 0? If anything, you shouldn't set `t0` at all, because the default behavior is to set it to the systems _original_ starting time. If you e.g., had a periodically forced dynamical system, this t0 would have meaning. Is there a particular reason to enforce `t0 = 0` here? I don't see it from the docstring.

Reply:
I don't remember why I set this as 0. I leave the default initial time.